attestation-service/docs/grpc-as.md

```diff
@@ -2,15 +2,62 @@
 
 `grpc-as` is an Attestation Service application based on gRPC protocol.
 
-## API
+Now the following types of evidence are supported:
+- `azsnpvtpm`: Azure SNP vTPM
+- `sev`: AMD SEV (Not implemented)
+- `sgx`: Intel SGX
+- `snp`: AMD SNP
+- `tdx`: Intel TDX
+- `cca`: Arm CCA
+- `sample`: A fake platform. Only for test and sample
+- `csv`: Hygon CSV
+- `aztdxvtpm`: Azure TDX vTPM (Not implemented)
 
-gRPC Attestation Service provides gRPC endpoints which is defined in [protobuf](../protos/attestation.proto).
+## Quick Start
 
-## Usage
+Users can use a [community version of gRPC CoCoAS image](https://github.com/confidential-containers/trustee/pkgs/container/staged-images%2Fcoco-as-grpc) to verify attestation reports.
 
-Here are the steps of building and running gRPC Attestation Service:
+```shell
+# run gRPC CoCoAS server locally
+docker run -d \
+  -v /etc/sgx_default_qcnl.conf:/etc/sgx_default_qcnl.conf \ # this qcnl config is used when verifying SGX/TDX quotes
```

Comment:
Where is this file coming from?

Reply:
This is the dcap configuration file that is used for intel platforms. if we install `libsgx-dcap-default-qpl` and `libsgx-dcap-quote-verify`, we can see a `/etc/sgx_default_qcnl.conf` file.

This is tricky because the file defines an address of a server storing endorsements of TEE keys, named PC(C)S. Different CSPs often provide their own PCCS service thus it is hard to have a common address. Thus it is hard to provide a workable file for all cases.

I am trying to assume that users of intel platform can quickly know what does this file mean. Do you think I should add an example file or not?

might also get some information from @mythi

Reply:
Ok I will test with the package, but if not, I am not sure how can someone really test this document. Also this is trustee so it is supposed to run anywhere agnostic of the hardware, isn't it?

Reply:
FWIW, in Trustee CI we use Intel PCS directly but cache the results so that's also an option.

Reply:
I only found that we change the `collateral_service` is changed to `https://api.trustedservices.intel.com/sgx/certification/v4/` [here](https://github.com/confidential-containers/trustee/blob/main/kbs/test/Makefile#L41), but `pccs_url` not set. Is that enough?

Reply:
It's enough for the functionality but that config must only be used when the results are cached and re-used for subsequent runs (which is what CI does)

Reply:
I was able to use the following config as is from this [MSFT webpage](https://learn.microsoft.com/en-us/azure/security/fundamentals/trusted-hardware-identity-management#how-do-i-use-intel-qpl-with-trusted-hardware-identity-management) and it worked nicely. I was wondering if we can use some canonical configuration like this which will work regardless of where you run the AS? Is there a default Intel service or can we rely on this Azure configuration or Alibaba configuration?

```json
{ 
    "pccs_url": "https://global.acccache.azure.net/sgx/certification/v3/", 
    "use_secure_cert": true, 
    "collateral_service": "https://global.acccache.azure.net/sgx/certification/v3/",  
    "pccs_api_version": "3.1", 
    "retry_times": 6, 
    "retry_delay": 5, 
    "local_pck_url": "http://169.254.169.254/metadata/THIM/sgx/certification/v3/",
    "pck_cache_expire_hours": 24, 
    "verify_collateral_cache_expire_hours": 24, 
    "custom_request_options": { 
        "get_cert": { 
            "headers": { 
                "metadata": "true" 
            }, 
            "params": { 
                "api-version": "2021-07-22-preview" 
            } 
        } 
    } 
}
```

Reply:
@surajssd I linked to a default one that we use in e2e CI test. Also give some guide links for users to configure cloud provider's PCCS. ptal  https://github.com/confidential-containers/trustee/pull/365/files#diff-f97bf4b195e75c833eb0ba1cb82dcb1f43deec96699eb2bf98b37ad7975fe710R28-R33